### DISCUSSION TITLE
Feature request: specify displays by name or by position relative to main

### DISCUSSION THREAD

**Discussion by gimbo** [rxhanson/RectanglePro-Community#369] [Ideas]
## Short version

Allow displays to be specified by device name ("built-in", "BenQ LCD", etc.) or by their position relative to the main display.

## Long version

Right now, displays can be specified like this when setting up custom positions or app layouts:

* Unchanged
* Next / Previous — relative to current window position
* Absolute numerical (1, 2, 3, ...) — according to the tooltip "The display number is determined by x and y coordinates of the display at the time of execution."

I often want to position things on particular displays (so not Unchanged or Next or Previous) but find the x/y-based fixed numbers limiting (see below for motivating example).

As such, it would be useful to also be able to use display names, e.g.:

* Built-in Display
* BenQ LCD
* iPad (Sidecar)

and/or (ideally "and"!) positions relative to "main", e.g.:

* Main display — the one with the menubar on it
* First display below the main display
* Second display to the right of main display
* Etc.

Keyboard Maestro lets you do something similar; I've used that in the past for window management but migrated to RP as it's considerably better — however, I find myself missing this.

By way of motivation: I work with several different display layouts depending on physical context (home, office, coffee shop, etc.) but I have some consistent patterns of use which can't currently be reflected nicely via RP. E.g.:

* The biggest display in front of me is always main (this is handled by macOS display settings of course); then I want:

    - `CapsLock-S` centres a window on main
    
	- IDEs go to main by default.
	
   ... but that's not currently consistently possible because e.g. at home the main display is display 2 (there's another display to its left) but in a coffee shop or the office it's always display 1.

* I often use my iPad as a Sidecar display; at home it sits below the main display (an external monitor) but in a coffee shop it sits to the right (main would be the laptop's built-in display); in either case, I want:

    - `CapsLock-2` maximises a window on it.
    
    - iTerm launches to it, maximised.
    
   ... but again this isn't consistently possible right now because in my home office, it's manifesting as display 4, but in the coffee shop it's display 2.

In general I tend to have an ordering of displays in my head from most important to least, which doesn't necessarily (or ever?) map to the x/y-defined ordering RP currently uses; and in general I sometimes want to use particular hardware for a particular purpose.

As such, I think this would be a very useful addition to RP's abilities, opening up a number of use cases which aren't currently possible.

Obviously it needs some thought around fallback; e.g. what to do when "second display to the left of main" doesn't actually exist? Possible answers:

* Current, or main if not already open.
* Main
* Next-closest display to the left of main (so "first display to the left of main" if there is one, otherwise main).

**Comment by rxhanson** [comment]
Thanks for the detailed feedback. I see what you mean, and agree that the multi-monitor support could be improved. I'll add this one to the backlog, but can't make any guarantees on when I'd be able to implement it.
